workers/unity/Assets/Playground/Scripts/EntityInitialization/EntityGameObjectCreator.cs

```diff
@@ -1,46 +1,66 @@
-using System;
 using System.Collections.Generic;
-using Improbable.Worker;
-using Unity.Entities;
+using System.IO;
+using Generated.Improbable;
+using Generated.Improbable.Transform;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.GameObjectRepresentation;
 using UnityEngine;
+using Transform = Generated.Improbable.Transform.Transform;
 
 namespace Playground
 {
-    public class EntityGameObjectCreator
+    public class EntityGameObjectCreator : IEntityGameObjectCreator
     {
-        private readonly World world;
-        private readonly Dictionary<string, GameObject> cachedPrefabs;
+        private readonly Dictionary<string, Dictionary<string, GameObject>> cachedPrefabsPerWorkerType
```

Comment:
If we aren't making this worker specific can we just make this a single map?

Reply:
Just seen that we are doing two worker types. Commented below.

Reply:
We are making it optionally worker specific, see code below - we first check in the path `Prefabs\<WorkerName>\<EntityName>`.